src/pulsar_socket.erl

```diff
@@ -107,6 +134,18 @@ send_batch_message_packet(Sock, Topic, Messages, SequenceId, ProducerId, Produce
     pulsar_metrics:send(Topic, length(Messages)),
     Mod:send(Sock, pulsar_protocol_frame:send(SendCmd, MsgMetadata, BatchMsg)).
 
+send_batch_message_packet_async(SockPid, Topic, Messages, SequenceId, ProducerId, ProducerName, Opts) ->
+    Req =
+        #send_batch{ topic = Topic
```

Comment:
encode the batch by caller so to send potentially large binary ref instead of copy a large/deep term.

Reply:
:checkered_flag: